File: backend/src/main/java/com/example/demo/model/Shop.java
package com.example.demo.model;


import javax.persistence.*;
import java.io.Serializable;

@Entity
@Table(name = "shop")
public class Shop implements Serializable {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private String name;
    private String address;
    private Double lat;
    private Double lng;
    private Boolean withdrawn;
    private String brand;


    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public String getAddress() {
        return address;
    }

    public void setAddress(String address) {
        this.address = address;
    }

    public Double getLat() {
        return lat;
    }

    public void setLat(Double lat) {
        this.lat = lat;
    }

    public Double getLng() {
        return lng;
    }

    public void setLng(Double lng) {
        this.lng = lng;
    }

    public Boolean getWithdrawn() {
        return withdrawn;
    }

    public void setWithdrawn(Boolean withdrawn) {
        this.withdrawn = withdrawn;
    }

    public String getBrand() {
        return brand;
    }

    public void setBrand(String brand) {
        this.brand = brand;
    }
}

File: web/src/app/models/shop.ts
import { GeoJson } from "./map";

export class Shop {
  _id?: number;
  name: string;
  lng: number;
  lat: number;
  brand: string;
  address: string;
  tags: string[];
  withdrawn?: boolean;

  constructor(options: {
    _id?: number,
    name: string,
    lng: number,
    lat: number,
    brand: string,
    address: string,
    tags: string[],
    withdrawn?: boolean,
  }) {
    if ("_id" in options)
      this._id = options._id;
    this.name = options.name;
    this.lng = options.lng;
    this.lat = options.lat;
    this.brand = options.brand;
    this.address = options.address;
    this.tags = options.tags;
    if ("withdrawn" in options)
      this.withdrawn = options.withdrawn;
  }

  toJSON() {
    return Object.assign({}, this);
  }

  toGeoJSON() {
    return new GeoJson(
      [this.lng, this.lat],
      {
        name: this.name,
        brand: this.brand,
        address: this.address,
        tags: this.tags,
        _id: this._id
      }
    );
  }

}
